programs/nft-voter/src/instructions/update_voter_weight_record.rs

```diff
@@ -6,6 +6,11 @@ use itertools::Itertools;
 /// Updates VoterWeightRecord to evaluate governance power for non voting use cases: CreateProposal, CreateGovernance etc...
 /// This instruction updates VoterWeightRecord which is valid for the current Slot and the given target action only
 /// and hance the instruction has to be executed inside the same transaction as the corresponding spl-gov instruction
+///
+/// Note: UpdateVoterWeight is not cumulative the same way as CastNftVote and hence voter_weight for non voting scenarios
+/// can only be used with max 5 NFTs due to Solana transaction size limit
+/// It could be supported in future version by introducing bookkeeping accounts to track the NFTs
+/// which were already used to calculate the total weight
```

Comment:
How about making updates impossible on nft votes?  That way, nft voters must relinquish their vote in order to update.

What happens when someone casts a lot of NFT votes and then only updates.  For example, I use 5 transactions to cast 25 nft votes, but then I update only 5 of them -- does that make me lose out on my previous voting power?

Reply:
Updates are not possible and yes NFT voters must relinquish their vote if they want to update their vote weight when for example they got more NFTs.

Both `CastNftVote` and `UpdateVoterWeightRecord` instructions  update `VoterWeightRecord` account which is then used by `spl-gov` as the source of truth for `voter_weight`. However they set different `voter_weight_action` which determines what the weight can be used for https://github.com/solana-labs/governance-program-library/blob/8a599cd9d0aeeeb04ccd526665774b30dd444c63/programs/nft-voter/src/state/voter_weight_record.rs#L8 

For that reason we can call `CastNftVote` instruction multiple times and it creates the NFT vote markers accounts `NftVoteRecord` which prevent double voting by the same NFT and each instruction also keeps accumulating the total `voter_weight` which is then passed to `spl-gov`
This is the `Update`, `Update`, .... `Commit` pattern used for distributed transactions 

If we wanted to implement the accumulative logic  for `UpdateVoterWeightRecord` we would also have to create the markers and this is what I left for future versions if we find it's needed.